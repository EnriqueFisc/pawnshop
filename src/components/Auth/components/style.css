.Login__form-container {
    width: 70%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.login__login-input {
    width: 100%;
    margin: 5px 5px;
    padding: 10px 10px;
    outline: none;
    border: 1px solid #919191;
    color: #919191;
    border-radius: 5px;
}

.login__login-button {
    width: 100%;
    margin: 5px 5px;
    padding: 10px 10px;
    background-color: #2162E5;
    outline: none;
    border: 0;
    color: white;
    cursor: pointer;
    border-radius: 5px;
}
.login__login-button:hover {
    background-color: #064ddc;
}
.login__login-button:active {
    background-color: #2c67dc;
}

@media (max-width: 700px) {
    .Login__form-container {
        width: 90%;
    }
}