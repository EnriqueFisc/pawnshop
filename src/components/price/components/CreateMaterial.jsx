import React, { useState } from "react";
import { Notification } from "../../Notifications/Notification";

export const CreateMaterial = () => {
  const [formValues, setformValues] = useState({
    materialId: "",
    gramsPrice: 0,
    name: "",
  });
  const [notification, setNotification] = useState({
    load: false,
    data: "",
  });

  const { materialId, gramsPrice, name } = formValues;

  const handleOnSubmit = (e) => {
    e.preventDefault();
    const token = localStorage.getItem("token");
    fetch("http://167.71.17.236:3000/materials/create", {
      method: "POST",
      body: JSON.stringify(formValues),
      headers: {
        "Content-type": "application/json",
        Access: token,
      },
    })
      .then((response) => response.json())
      .then((data) => {
        if (data.ok) {
          setNotification({
            load: true,
            data: data.msg,
            ok: true,
          });
          return;
        }

        setNotification({
          load: true,
          data: data.msg,
          ok: false,
        });
      });
  };

  const handleOnChange = (e) => {
    setformValues((prevFormValues) => ({
      ...prevFormValues,
      [e.target.name]: e.target.value,
    }));
  };
  return (
    <form className="prices__form-container" onSubmit={handleOnSubmit}>
      {notification.load && (
        <Notification
          msg={notification.data}
          ok={notification.ok}
          setNotification={setNotification}
        />
      )}
      <h3>Crear nuevo producto</h3>
      <input
        className="prices__form-inputs"
        type="text"
        name="materialId"
        value={materialId}
        placeholder="Ingresa el codigo del material"
        onChange={handleOnChange}
      />
      <input
        className="prices__form-inputs"
        type="text"
        name="name"
        value={name}
        placeholder="Ingresa el nombre del material"
        onChange={handleOnChange}
      />
      <input
        className="prices__form-inputs"
        type="number"
        name="gramsPrice"
        value={gramsPrice}
        placeholder="Ingresa el precio por gramo"
        onChange={handleOnChange}
      />
      <button className="prices__form-button">Crear</button>
    </form>
  );
};
