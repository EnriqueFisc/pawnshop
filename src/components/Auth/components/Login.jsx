import React, { useState } from "react";
import "./style.css";
import { useNavigate } from "react-router-dom";

export const LoginFormContaine = () => {
  const navigate = useNavigate();
  const [formValues, setformValues] = useState({
    email: "",
    password: "",
  });
  const [notification, setNotification] = useState({
    load: false,
    data: "",
  });

  const { email, password } = formValues;

  const handleOnSubmit = (e) => {
    e.preventDefault();
    console.log("submit", formValues);

    fetch("http://localhost:4000/users/auth", {
      method: "POST",
      body: JSON.stringify(formValues),
      headers: {
        "Content-type": "application/json",
      },
    })
      .then((response) => response.json())
      .then((data) => {
        if (data.ok) {
          localStorage.setItem("token", data.token);
          navigate("/prices");
          return;
        }

        setNotification({
          load: true,
          data: data.msg,
        });
      });
  };

  const handleOnChange = (e) => {
    setformValues((prevFormValues) => ({
      ...prevFormValues,
      [e.target.name]: e.target.value,
    }));
  };

  return (
    <form className="Login__form-container" onSubmit={handleOnSubmit}>
      {notification.load && <div>{notification.data}</div>}
      <input
        className="login__login-input"
        type="email"
        name="email"
        value={email}
        placeholder="Ingrese su correo electrónico"
        onChange={handleOnChange}
        autoComplete="off"
      />
      <input
        className="login__login-input"
        type="password"
        name="password"
        value={password}
        placeholder="Ingrese su contraseña"
        onChange={handleOnChange}
      />
      <button className="login__login-button">Login</button>
    </form>
  );
};
