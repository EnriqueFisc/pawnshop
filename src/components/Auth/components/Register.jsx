import React, { useState } from "react";
import "./style.css";
import { useNavigate } from "react-router-dom";

export const RegisterFormContaine = () => {
  const navigate = useNavigate();
  const [formValues, setformValues] = useState({
    email: "",
    password: "",
    passwordConfirm: "",
    name: "",
    lastname: "",
  });
  const [notification, setNotification] = useState({
    load: false,
    data: "",
  });

  const { email, password, passwordConfirm, name, lastname } = formValues;

  const handleOnSubmit = (e) => {
    e.preventDefault();
    if (password !== passwordConfirm) {
      setNotification({
        load: true,
        data: "Las contrase침as deben ser iguales",
      });
      return;
    }
    for (const value in formValues) {
      if (!formValues[value]) {
        setNotification({
          load: true,
          data: "Por favor llena todos los campos",
        });
        return;
      }
    }
    fetch("http://167.71.17.236:4000/users", {
      method: "POST",
      body: JSON.stringify(formValues),
      headers: {
        "Content-type": "application/json",
      },
    })
      .then((response) => response.json())
      .then((data) => {
        if (data.ok) {
          localStorage.setItem("token", data.token);
          navigate("/prices");
          return;
        }

        setNotification({
          load: true,
          data: data.msg,
        });
      });
  };

  const handleOnChange = (e) => {
    setformValues((prevFormValues) => ({
      ...prevFormValues,
      [e.target.name]: e.target.value,
    }));
  };

  return (
    <form className="Login__form-container" onSubmit={handleOnSubmit}>
      {notification.load && <div>{notification.data}</div>}
      <input
        className="login__login-input"
        type="text"
        name="name"
        value={name}
        placeholder="Ingrese sus nombres"
        onChange={handleOnChange}
        autoComplete="off"
      />
      <input
        className="login__login-input"
        type="text"
        name="lastname"
        value={lastname}
        placeholder="Ingrese sus apellidos"
        onChange={handleOnChange}
        autoComplete="off"
      />
      <input
        className="login__login-input"
        type="email"
        name="email"
        value={email}
        placeholder="Ingrese su correo electr칩nico"
        onChange={handleOnChange}
        autoComplete="off"
      />
      <input
        className="login__login-input"
        type="password"
        name="password"
        value={password}
        placeholder="Ingrese su contrase침a"
        onChange={handleOnChange}
      />
      <input
        className="login__login-input"
        type="password"
        name="passwordConfirm"
        value={passwordConfirm}
        placeholder="Confirme su contrase침a"
        onChange={handleOnChange}
      />
      <button className="login__login-button">Registrarme</button>
    </form>
  );
};
