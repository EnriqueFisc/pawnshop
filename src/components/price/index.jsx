import React, { useState } from "react";
import "./styles.css";
import { CreateMaterial } from "./components/CreateMaterial";
import { Notification } from "../Notifications/Notification";

export const Prices = () => {
  const [formValues, setformValues] = useState({
    materialId: "",
    grams: 0,
  });
  const [price, setPrice] = useState("");
  const [notification, setNotification] = useState({
    load: false,
    data: "",
  });

  const { materialId, grams } = formValues;

  const handleOnSubmit = (e) => {
    e.preventDefault();
    const token = localStorage.getItem("token");

    fetch("http://167.71.17.236:3000/materials/price", {
      method: "POST",
      body: JSON.stringify(formValues),
      headers: {
        "Content-type": "application/json",
        Access: token,
      },
    })
      .then((response) => response.json())
      .then((data) => {
        if (data.ok) {
          setPrice(data.price);
          return;
        }

        setNotification({
          load: true,
          data: data.msg,
        });
      });
  };

  const handleOnChange = (e) => {
    setformValues((prevFormValues) => ({
      ...prevFormValues,
      [e.target.name]: e.target.value,
    }));
  };
  return (
    <div className="prices__window-container">
      {notification.load && (
        <Notification
          msg={notification.data}
          setNotification={setNotification}
        />
      )}
      <div className="prices__container">
        <div className="prices__calculation-container">
          <h3>Resultado</h3>
          <div className="prices__calculation">
            <div className="prices__calculation-name">
              <p>Monto del prestamo: {`${price}$`}</p>
            </div>
          </div>
        </div>
        <form className="prices__form-container" onSubmit={handleOnSubmit}>
          <h3>Ingresa productos para calcular</h3>
          <div className="prices__form-material-data">
            <input
              className="prices__form-inputs"
              type="text"
              name="materialId"
              value={materialId}
              placeholder="Ingresa el codigo del material"
              onChange={handleOnChange}
            />
            <input
              className="prices__form-inputs"
              type="number"
              name="grams"
              value={grams}
              placeholder="Ingresa los gramos"
              onChange={handleOnChange}
            />
          </div>
          <button className="prices__form-button">Calcular Precio</button>
        </form>
        <CreateMaterial />
      </div>
    </div>
  );
};
