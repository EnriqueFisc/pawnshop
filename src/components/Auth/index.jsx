import React, { useState } from "react";
import { LoginFormContaine } from "./components/Login";
import "./style.css";
import { RegisterFormContaine } from "./components/Register";

export const Auth = () => {
  const [isLogin, setIsLogin] = useState(true);

  const handleOnCLickChangeForm = () => {
    setIsLogin((prevState) => !prevState);
  };

  return (
    <div className="login__container">
      <div className="login__container-leftside">
        {isLogin ? (
          <div className="loginfrom__container">
            <img className="image__logo" src="/public/nmp_log.png" alt="logo" />
            <div className="loginfrom__container-title">
              <h2>Inicia Sesi√≥n</h2>
            </div>
            <LoginFormContaine />
            <button
              className="login__login-button-register"
              onClick={handleOnCLickChangeForm}
            >
              Registrate
            </button>
          </div>
        ) : (
          <>
            <div className="loginfrom__container">
              <img
                className="image__logo"
                src="/public/nmp_log.png"
                alt="logo"
              />
              <div className="loginfrom__container-title">
                <h2>Registrate</h2>
              </div>
              <RegisterFormContaine />
              <button
                className="login__login-button-register"
                onClick={handleOnCLickChangeForm}
              >
                Ya tengo una cuenta
              </button>
            </div>
          </>
        )}
      </div>
      <div className="login__container-rightside"></div>
    </div>
  );
};
