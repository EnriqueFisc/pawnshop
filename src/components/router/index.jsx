import React, { useEffect } from "react";
import {
  BrowserRouter as Router,
  Route,
  Link,
  Routes,
  useNavigate,
} from "react-router-dom";
import { Auth } from "../Auth";
import { Prices } from "../price";
import { ContainerRoutes } from "./Router";
export const MainRouter = () => {
  return (
    <Router>
      <ContainerRoutes />
    </Router>
  );
};
