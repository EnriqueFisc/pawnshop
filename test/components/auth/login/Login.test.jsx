import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { LoginFormContaine } from "../../../../src/components/Auth/components/Login";
import { BrowserRouter } from "react-router-dom";

describe("Test in LoginFormContaine Component", () => {
  test("should match with snapshot", () => {
    const component = render(
      <BrowserRouter>
        <LoginFormContaine />
      </BrowserRouter>
    );
    expect(component.container).toMatchSnapshot();
  });
});
