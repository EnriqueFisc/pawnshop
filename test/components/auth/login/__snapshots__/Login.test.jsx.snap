// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test in LoginFormContaine Component > should match with snapshot 1`] = `
<div>
  <form
    class="Login__form-container"
  >
    <input
      autocomplete="off"
      class="login__login-input"
      name="email"
      placeholder="Ingrese su correo electrónico"
      type="email"
      value=""
    />
    <input
      class="login__login-input"
      name="password"
      placeholder="Ingrese su contraseña"
      type="password"
      value=""
    />
    <button
      class="login__login-button"
    >
      Login
    </button>
  </form>
</div>
`;
